src/Plugin/Deriver/EntityAliasTypeDeriver.php

```diff
@@ -0,0 +1,27 @@
+<?php
+/**
+ * @file
+ * Contains \Drupal\pathauto\Plugin\Deriver\EntityAliasTypeDeriver.
+ */
+
+namespace Drupal\pathauto\Plugin\Deriver;
+
+use Drupal\Core\Plugin\Context\ContextDefinition;
+use Drupal\ctools\Plugin\Deriver\EntityDeriverBase;
+
+class EntityAliasTypeDeriver extends EntityDeriverBase {
+  public function getDerivativeDefinitions($base_plugin_definition) {
+    foreach ($this->entityManager->getDefinitions() as $entity_type_id => $entity_type) {
+      if ($entity_type->hasLinkTemplate('canonical')) {
```

Comment:
I'm not sure about the connection between this, the canonical link template and the path field. Without a path field, this won't do much. See pathauto_entity_base_field_info() which adds it for user. If we do this automatically, maybe we should also add the base field for those entity types? I was also wondering about a checkboxes form element on the settings page that allows to add such a field. Maybe base field + plugin should be separate, always add the plugin but make adding the base field configurable? The plugin has fairly little overhead now if there are no config entities.

Do we support non-content entities now with this? Not many config entities do have one, none in core afaik, but e.g. search api Index/Server do.

Last, I think this exposes plugin ID's like canonical_entities:node, but the plugin property on entities talks about just node, user, etc.